### PR TITLE
Parameter type leftovers

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179 
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by derrabus** [comment]
Status: Needs Work

File: src/Symfony/Component/Console/Question/Question.php [hunk 1]
```diff
@@ -220,7 +215,7 @@ public function getValidator()
      *
      * @throws InvalidArgumentException in case the number of attempts is invalid
      */
-    public function setMaxAttempts($attempts)
+    public function setMaxAttempts(?int $attempts)
```
Comment (Tobion) [inline src/Symfony/Component/Console/Question/Question.php]: you could remove the param doc

Comment (tigitz) [reply src/Symfony/Component/Console/Question/Question.php]: @derrabus the now merged https://github.com/symfony/symfony/issues/33150 could help to clean those automatically

**Comment by derrabus** [comment]
Status: Needs Review

File: src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AbstractPhpFileCacheWarmer.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ protected function warmUpPhpArrayAdapter(PhpArrayAdapter $phpArrayAdapter, array
     /**
      * @internal
      */
-    final protected function ignoreAutoloadException($class, \Exception $exception)
+    final protected function ignoreAutoloadException(string $class, \Exception $exception)
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AbstractPhpFileCacheWarmer.php]: some

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php [hunk 3]
```diff
@@ -26,6 +27,9 @@
  */
 abstract class AbstractConfigCommand extends ContainerDebugCommand
 {
+    /**
+     * @param OutputInterface|StyleInterface $output
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: changes

File: src/Symfony/Component/BrowserKit/Cookie.php [hunk 4]
```diff
@@ -213,6 +213,8 @@ private static function parseDate($dateValue)
         if (false !== $date = date_create($dateValue, new \DateTimeZone('GMT'))) {
             return $date->format('U');
         }
+
+        return null;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/BrowserKit/Cookie.php]: are

File: src/Symfony/Component/BrowserKit/CookieJar.php [hunk 5]
```diff
@@ -103,10 +103,9 @@ public function clear()
     /**
      * Updates the cookie jar from a response Set-Cookie headers.
      *
-     * @param array  $setCookies Set-Cookie headers from an HTTP response
-     * @param string $uri        The base URL
+     * @param string[] $setCookies Set-Cookie headers from an HTTP response
```
Comment (nicolas-grekas) [inline src/Symfony/Component/BrowserKit/CookieJar.php]: for

File: src/Symfony/Component/Config/Resource/ClassExistenceResource.php [hunk 6]
```diff
@@ -128,7 +128,7 @@ public function __sleep(): array
      *
      * @internal
      */
-    public static function throwOnRequiredClass($class, \Exception $previous = null)
+    public static function throwOnRequiredClass(string $class, \Exception $previous = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Resource/ClassExistenceResource.php]: a

File: src/Symfony/Component/Console/Question/Question.php [hunk 7]
```diff
@@ -130,14 +130,11 @@ public function getAutocompleterValues()
     /**
      * Sets values for the autocompleter.
      *
-     * @param iterable|null $values
-     *
      * @return $this
      *
-     * @throws InvalidArgumentException
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Question/Question.php]: lower

File: src/Symfony/Component/Console/Question/Question.php [hunk 8]
```diff
@@ -150,10 +147,8 @@ public function setAutocompleterValues($values)
             $callback = static function () use ($values, &$valueCache) {
                 return $valueCache ?? $valueCache = iterator_to_array($values, false);
             };
-        } elseif (null === $values) {
-            $callback = null;
         } else {
-            throw new InvalidArgumentException('Autocompleter values can be either an array, "null" or a "Traversable" object.');
+            $callback = null;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Question/Question.php]: branch

File: src/Symfony/Component/Messenger/MessageBus.php [hunk 9]
```diff
@@ -39,7 +39,7 @@ public function __construct(iterable $middlewareHandlers = [])
                 private $middlewareHandlers;
                 private $cachedIterator;
 
-                public function __construct($middlewareHandlers)
+                public function __construct(\Traversable $middlewareHandlers)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/MessageBus.php]: isn't

File: src/Symfony/Component/Process/Process.php [hunk 10]
```diff
@@ -979,36 +979,32 @@ public function getIdleTimeout()
     }
 
     /**
-     * Sets the process timeout (max. runtime).
+     * Sets the process timeout (max. runtime) in seconds.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: it?

File: src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AbstractPhpFileCacheWarmer.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/CacheWarmer/AbstractPhpFileCacheWarmer.php]: #33198

File: src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php (hunk 3 above)
Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php]: #33197

File: src/Symfony/Component/BrowserKit/Cookie.php (hunk 4 above)
Comment (derrabus) [reply src/Symfony/Component/BrowserKit/Cookie.php]: #33197

File: src/Symfony/Component/BrowserKit/CookieJar.php (hunk 5 above)
Comment (derrabus) [reply src/Symfony/Component/BrowserKit/CookieJar.php]: #33197

File: src/Symfony/Component/Console/Question/Question.php (hunk 7 above)
Comment (derrabus) [reply src/Symfony/Component/Console/Question/Question.php]: The code that throws the exception is removed by this PR. Lower branches still have it.

File: src/Symfony/Component/Console/Question/Question.php (hunk 8 above)
Comment (derrabus) [reply src/Symfony/Component/Console/Question/Question.php]: I'm removing a type check here, that was obsoleted by the new parameter type declaration. Lower branches still need the type-check.

File: src/Symfony/Component/Messenger/MessageBus.php (hunk 9 above)
Comment (derrabus) [reply src/Symfony/Component/Messenger/MessageBus.php]: Added to #33198.

File: src/Symfony/Component/Process/Process.php (hunk 10 above)
Comment (derrabus) [reply src/Symfony/Component/Process/Process.php]: All right: #33221

**Comment by nicolas-grekas** [comment]
rebase unlocked.

File: src/Symfony/Component/Config/Resource/ClassExistenceResource.php (hunk 6 above)
Comment (derrabus) [reply src/Symfony/Component/Config/Resource/ClassExistenceResource.php]: I'm preparing a PR for `@internal` methods and will pick this one up there as well.

**Comment by derrabus** [comment]
Ready.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Translation/Translator.php [hunk 11]
```diff
@@ -422,11 +420,9 @@ protected function computeFallbackLocales($locale)
...
      * Asserts that the locale is valid, throws an Exception if not.
      *
-     * @param string $locale Locale to tests
-     *
      * @throws InvalidArgumentException If the locale contains invalid characters
      */
-    protected function assertValidLocale($locale)
+    protected function assertValidLocale(?string $locale)
     {
         if (1 !== preg_match('/^[a-z0-9@_\\.\\-]*$/i', $locale)) {
```
Comment (Tobion) [inline src/Symfony/Component/Translation/Translator.php]: preg_match should not be a called with null.

Comment (Tobion) [reply src/Symfony/Component/Translation/Translator.php]: Also I assume the nullable locale comes from `setLocale`. But according to https://github.com/symfony/symfony/blob/4.4/src/Symfony/Contracts/Translation/LocaleAwareInterface.php it's not nullable. Seems like we need to fix the interface then.

Comment (nicolas-grekas) [reply src/Symfony/Component/Translation/Translator.php]: let's take this as an opportunity to remove the `1 !==` comparison, we don't do this in the codebase when using preg_*

Comment (Tobion) [reply src/Symfony/Component/Translation/Translator.php]: The translator contracts have no type declarations. Have those been skipped on purpose?

Comment (derrabus) [reply src/Symfony/Component/Translation/Translator.php]: The contracts need to support php 7.1. Adding parameter type declarations there would be a breaking change.

Comment (derrabus) [reply src/Symfony/Component/Translation/Translator.php]: The unit tests explicitly list `null` as a valid value for `$locale`: https://github.com/symfony/symfony/blob/ef0119a2d40f1a5930b4fd9146b1569b17d849ea/src/Symfony/Component/Translation/Tests/TranslatorTest.php#L489

Comment (derrabus) [reply src/Symfony/Component/Translation/Translator.php]: #33247

**Comment by derrabus** [comment]
Rebased. PR is ready.

**Review by Tobion — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**